src/main/java/stats/KMeans.java

```diff
@@ -0,0 +1,162 @@
+package stats;
+
+import stats.utils.Cluster;
+import stats.utils.Point;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Implementation of K-Means Clustering Algorithm.
+ */
+public class KMeans {
+
+    public void init() {
+
+        /**
+         * Create clusters and set random centroids.
+         */
+
+        for (int i = 0; i < k; ++i) {
+            Cluster cluster = new Cluster(i);
+            Point centroid = Point.createRandomPoint(MIN_COORDINATE, MAX_COORDINATE);
+            cluster.setCentroid(centroid);
+            clusters.add(cluster);
+        }
+
+    }
+
+    public void calculate() {
+        int iterations = 0;
+
+        while (iterations <= maxIterations) {
+
+            clearClusters();
+
+            List<Point> lastCentroids = getCentroids();
+
+            assignCluster();
+
+            calculateCentroids();
+
+            iterations++;
+
+            List<Point> currentCentroids = getCentroids();
+
+            double distance = 0;
+            for (int i = 0; i <= lastCentroids.size() ; ++i) {
+                distance += Point.distance(lastCentroids.get(i), currentCentroids.get(i));
+            }
+            System.out.println("#################");
```

Comment:
For something that needs 5000 iterations that will be quite an overwhelming point plus it will slow down thins...this I/O should be done conditionally

Reply:
I can remove it.
Just put it there for logging issues.